### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by shepwalker**

### CODE REVIEW

File: run.py [hunk 1]
```diff
@@ -1,6 +1,12 @@
 # This Python file uses the following encoding: utf-8
 import os
+import sys
+import logging
```
Comment (jmduke) [inline run.py]: nitpick: organize alphabetically

File: run.py [hunk 2]
```diff
@@ -15,13 +21,40 @@
 
 app = Flask(__name__)
 
-SPREADSHEET_URL = os.getenv('CORGI_URL', '')
+# logging nonsense
+logpath = os.getenv('CORJI_LOG_PATH', './logs')
```
Comment (jmduke) [inline run.py]: break out all this stuff into a separate file (`logging.py` or some shit) and just import ith ere

File: run.py [hunk 3]
```diff
@@ -15,13 +21,40 @@
...
+
+file_handler = TimedRotatingFileHandler(logpath + '/corji.log', 'midnight', 1)
```
Comment (jmduke) [inline run.py]: also make the logname an env var

File: run.py [hunk 4]
```diff
@@ -15,13 +21,40 @@
...
+file_handler.setFormatter(
+    logging.Formatter('%(asctime)s %(name)-12s %(levelname)-8s %(message)s'))
+file_handler.setLevel(logging.INFO)
+mainLogger = app.logger
```
Comment (jmduke) [inline run.py]: camelCaseIsGross

File: run.py [hunk 5]
```diff
@@ -15,13 +21,40 @@
...
+mainLogger.addHandler(file_handler)
```
Comment (jmduke) [inline run.py]: Oh god the actual API uses camelcase? still, lets avoid it in our vars

File: run.py [hunk 6]
```diff
@@ -15,13 +21,40 @@
...
+mainLogger.setLevel(logging.INFO)
+
+if app.debug:
+    local_handler = logging.StreamHandler(sys.stdout)
+    mainLogger.addHandler(local_handler)
 
+SPREADSHEET_URL = os.getenv('CORGI_URL', '')
```
Comment (jmduke) [inline run.py]: nitpick: i want to only grab the env vars once when possible, so import this value from data_source.py

File: run.py (hunk 2 above)
Comment (jmduke) [inline run.py]: And at the top add a lil comment saying what levels we're using for each thang

File: run.py (hunk 1 above)
Comment (shepwalker) [inline run.py]: done. pending commit

File: run.py (hunk 2 above)
Comment (shepwalker) [inline run.py]: done. pending commit

File: run.py (hunk 3 above)
Comment (shepwalker) [inline run.py]: done. pending commit

File: run.py (hunk 4 above)
Comment (shepwalker) [inline run.py]: done. pending commit

File: run.py (hunk 5 above)
Comment (shepwalker) [inline run.py]: approved tweet. sufficiently shaming.

File: run.py (hunk 6 above)
Comment (shepwalker) [inline run.py]: did an entire settings path. needs tweaking but is functional.

File: log.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import logging
+import os
+import sys
+#LOG LEVELS:
```
Comment (jmduke) [inline log.py]: use docstring:

"""
blashdoaisdqw


"""

File: log.py [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+from logging.handlers import (
+    TimedRotatingFileHandler
+)
+
+def setup_app_logger(app):
+	if not os.path.exists(app.config['CORJI_LOG_PATH']):
+		os.makedirs(app.config['CORJI_LOG_PATH'])
+	file_handler = TimedRotatingFileHandler(app.config['CORJI_LOG_PATH'] + '/' + app.config['CORJI_LOG_NAME'], 'midnight', 1)
+	file_handler.setFormatter(
+	logging.Formatter('%(asctime)s %(name)-12s %(levelname)-8s %(message)s'))
```
Comment (jmduke) [inline log.py]: this might just be the diff tool being dumb: is this at the same indentation level as the line above it?

File: log.py [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+	file_handler.setLevel(logging.INFO)
+	mainLogger = app.logger
```
Comment (jmduke) [inline log.py]: main_logger

File: run.py [hunk 10]
```diff
@@ -1,6 +1,11 @@
 # This Python file uses the following encoding: utf-8
+import logging
```
Comment (jmduke) [inline run.py]: remove unused imports

File: run.py [hunk 11]
```diff
@@ -12,16 +17,36 @@
...
+main_logger.debug("START: Spreadsheet URL defined: %s", SPREADSHEET_URL)
 # TODO: GLOBALS BAD.
 corgis = data_sources.load_from_spreadsheet(SPREADSHEET_URL)
+
+main_logger.debug("START: Starting to load Corjis into cache.")
 cache.put_in_local_cache(corgis)
+main_logger.debug("START: Completed Corji Cache loading")
 
 
+def logged_view(f):
```
Comment (jmduke) [inline run.py]: move this to log.py as well

**Pull request merged**